pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -23,6 +23,7 @@ import (
 	"context"
 	"encoding/json"
 
```

Comment:
No changes required in `datavolume-validate.go`?  There is a bunch of stuff in there for PVC source I believe?  The mutate webhook should be just about creating the token.  The validate webhook should do the bulk of the validation.

Reply:
So I was considering this but looking at PVC validation, it now allows the PVC to not exist (https://github.com/kubevirt/containerized-data-importer/pull/2306) altogether, which made me unsure if I want to go there.

Anyway, I plan to be making a couple of changes to validation in the sourceRef follow up
(https://github.com/akalenyu/containerized-data-importer/commits/snapshot-sourceref)
We can keep discussing this issue there if that is okay

This PR is huge as it is and I have been stepping outside of it quite a bit 
to ensure it can progress towards making it in (https://github.com/kubevirt/containerized-data-importer/pull/2559, https://github.com/kubevirt/containerized-data-importer/pull/2544),
so I would prefer if we could not block over this

Reply:
I think we should support snapshot not existing just like pvc.  Also, I get that sourceRef is orthogonal (as long as we fail/don't do anything now).  As well as sourceVolumeMode support.  But I think we should lock down `source.Snapshot` here.  Including all validation and support of snappshot not existing.

Reply:
WDYT about some basic validation like https://github.com/akalenyu/containerized-data-importer/commit/74c4d290d214f2e855c04011d7c121853b2803b0#diff-2082e0e19255fc87c811d8b4347d274b06376449f517cca2e34ebbbdcfaf6cee, would that work?

I just feel like anything more than that is an enhancement; correct me if I'm wrong,
validation is nice as it will reject stuff early on but eventual consistency will get us to where we want to be in the end